recommender/engines/base/base_neural_network.py

```diff
@@ -0,0 +1,45 @@
+# pylint: disable=too-few-public-methods, line-too-long
+"""Abstract neural network"""
+
+from abc import ABC
+from typing import Tuple
+from itertools import chain
+
+from torch.nn import Linear, BatchNorm1d, ReLU
+
+
+class BaseNeuralNetwork(ABC):
+    """Abstract Neural Network"""
+
+    @staticmethod
+    def _create_layers(
+        input_dim: int, layers_sizes: Tuple[int], output_dim: int = None
+    ):
+        """
+        Create neural network layers based on a provided tuple which contains the size of layers
+
+        Args:
+            input_dim: input dimensionality,
+            layers_sizes: a tuple of target layers sizes,
+            output_dim: output dimensionality.
+        """
+        # The first layer
+        layers = [
+            Linear(input_dim, layers_sizes[0]),
+            BatchNorm1d(layers_sizes[0]),
+            ReLU(),
+        ]
+        # The middle layers
+        layers += list(
+            chain.from_iterable(
+                [
+                    [Linear(n_size, next_n_size), BatchNorm1d(next_n_size), ReLU()]
```

Comment:
This is a good generalization although we should add the ability to either include or exclude BatchNorm (which should default to include) and also a way to choose the hidden activation function (which should default to ReLU)